v2/spiffex509/roots.go

```diff
@@ -0,0 +1,37 @@
+package spiffex509
+
+import (
+	"crypto/x509"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+)
+
+// Roots represents a set of X.509 roots, keyed by trust domain.
+type Roots struct{}
```

Comment:
How does this compare/contrast to spiffebundle.Set? Should methods to manipulate bundle/set membership instead live in spiffebundle?

I feel like I am missing something w.r.t. the spiffex509 and spiffejwt packages. The minting parts and private key handling in particular surprised me. Would you mind going into a little more detail about how you see these two packages fitting into the bigger picture?

Reply:
`spiffex509` and `spiffejwt` were intended to be low-level `x509` and `jwt` related code. I didn't want to take a dependency on spiffe bundles here or be proscriptive about how these packages obtained keys for validation, etc. hence the `spiffejwt.KeyStore` and `spiffex509.RootStore` interfaces, but maybe I'm missing the mark?